### PR TITLE
tests and documents access control

### PR DESCRIPTION
**Pull request by rudolfix**

### CODE REVIEW

File: contracts/AccessControl/RoleBasedAccessPolicy.sol [hunk 1]
```diff
@@ -6,7 +6,21 @@ import './AccessControlled.sol';
...
 
-contract RoleBasedAccessControl is
+/// @title access control based on Access Control Lists
+/// @dev Allows to assign a list addresses to a set of roles (n:n relation) and querying if such specific assignment exists.
+///     This assignment happens in two contexts:
+///         - contract context which allows to build a set of local permissions enforced for particular contract
+///         - global context which defines set of global permissions that apply to any contract using this RoleBasedAccessPolicy as Access Policy
+///     Permissions are cascading as follows
+///         - evaluate permission for given subject for given object (local)
+///         - evaluate permission for given subject for all objects (globall)
```
Comment (krzkaczor) [inline contracts/AccessControl/RoleBasedAccessPolicy.sol]: typo

File: contracts/AccessControl/RoleBasedAccessPolicy.sol [hunk 2]
```diff
@@ -6,7 +6,21 @@ import './AccessControlled.sol';
 import '../Reclaimable.sol';
 
 
-contract RoleBasedAccessControl is
+/// @title access control based on Access Control Lists
+/// @dev Allows to assign a list addresses to a set of roles (n:n relation) and querying if such specific assignment exists.
```
Comment (krzkaczor) [inline contracts/AccessControl/RoleBasedAccessPolicy.sol]: `list of addresses`?

File: migrations/2_deploy_contracts.js [hunk 3]
```diff
@@ -20,33 +20,33 @@ module.exports = function deployContracts(deployer, network, accounts) {
 
   deployer.then(async () => {
     console.log("AccessControl deployment...");
```
Comment (krzkaczor) [inline migrations/2_deploy_contracts.js]: rename to `AccessPolicy deployment`

File: test/AccessControl.js [hunk 4]
```diff
@@ -1,111 +1,825 @@
 import { expect } from "chai";
 import EvmError from "./helpers/EVMThrow";
 import { eventValue } from "./helpers/events";
```
Comment (krzkaczor) [inline test/AccessControl.js]: rename file to AccessPolicy

Comment (rudolfix) [reply test/AccessControl.js]: I'd keep old name. Many components are tested here, besides AccessPolicy implementation (like AccessControlled etc.)

**Pull request merged**